File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/UserInHomeBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class UserInHomeBuilder implements CommandBuilder, Closeable
{
    public UserInHomeBuilder()
    {
    }
    public final void close()
    {
    }

    @Override
    public Command create(ArrayList<Integer> commandInBytes)
    {
        Command command  = new UserInHomeCommand();
        return command;
    }
}

File: server/src/protocol/UserInHomeBuilder.cpp
#include "UserInHomeBuilder.h"
#include "UserInHomeCommand.h"

using namespace std;
using namespace communication;

UserInHomeBuilder::UserInHomeBuilder()
{}

UserInHomeBuilder::~UserInHomeBuilder()
{}

std::unique_ptr<Command> UserInHomeBuilder::create(const std::vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<UserInHomeCommand>();

    return move(command);
}
